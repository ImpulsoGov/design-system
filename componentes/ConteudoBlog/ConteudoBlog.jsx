import React from "react";
import { sanitize } from "../sanitize";

import style from "./ConteudoBlog.module.css"

const ConteudoBlog = ({
    titulo,
    texto,
    capa,
    autor
  }) => {
    const meses = ["Janeiro","Fevereiro","Março","Abril","Maio","Junho","Julho","Agosto","Setembro","Outubro","Novembro","Dezembro"]
    const mes = new Date(autor.data).getMonth()
    const ano = new Date(autor.data).getFullYear()

    return (
      <div className={style.ConteudoBlogContainer}>
        <div>
            <div 
              className={style.ConteudoBlogTitulo}
              dangerouslySetInnerHTML={{
                __html: sanitize(titulo),
              }}        
            ></div>
            <img 
              className={style.ConteudoBlogCapa}
              src={capa} />
        </div>
        <div className={style.ConteudoBlogAutor}>
            <img
                className={style.ConteudoBlogAutorAvatar}
                src={autor.avatar}
                alt={autor.avatar}
            ></img>
            <div>{autor.nome} <span className={style.ConteudoBlogData}>  ·  {meses[mes]} {ano}</span></div>
        </div>
        <div 
          className={style.ConteudoBlogTexto}
          dangerouslySetInnerHTML={{
            __html: sanitize(texto.replace('\"',"'").replace('"',"'")),
          }}        
        ></div>
      </div>
    )};
        
  export {ConteudoBlog};