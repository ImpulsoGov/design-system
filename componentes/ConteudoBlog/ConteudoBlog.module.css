@import url("https://fonts.googleapis.com/css2?family=Inter:wght@40;500&display=swap");
.ConteudoBlogContainer{
    background-color: #FFF;
    padding: 80px 80px 273px 80px;
    font-family: 'Inter';
    font-style: normal;
    font-weight: 400;
    display: flex;
    flex-direction: column;
    gap: 80px;
}
.ConteudoBlogTitulo{
    width: 70%;
    font-size: 64px;  
    margin-bottom: 80px;
}

.ConteudoBlogCapa{
    width: 100% ;
    height: auto;
    border-radius: 20px;
}
.ConteudoBlogTexto{
    text-align: justify;
    font-size: 24px;  
}
.ConteudoBlogTexto a{
    color: #2EB280;
    text-decoration: underline;
}
.ConteudoBlogTexto img{
    width: 100%;
    height: auto;
}
.ConteudoBlogAutor{
    display: flex;
    flex-direction: row;
    align-items: center;
    font-family: 'Inter';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    color: #1F1F1F;
    top:-30px;
}
.ConteudoBlogData{
    font-family: 'Inter';
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    color: #606E78;
}
.ConteudoBlogAutorAvatar{
    margin-right: 10px;
}
.ConteudoBlogTexto iframe{
    width: 100%;
}
/* SMALL */
@media screen and (max-width: 1023px) {
    .ConteudoBlogContainer{
        padding: 40px 40px 180px 40px;
        gap: 40px;
    }    
    .ConteudoBlogTitulo{
        font-size: 32px;  
        padding-left: 0;
        width: 100%;
        margin-bottom: 40px;
    }
    .ConteudoBlogCapa{
        border-radius: 10px;
    }    
    .ConteudoBlogTexto{
        font-size: 14px;  
    }
}
/* MEDIUM */
@media screen and (min-width: 1024px)and (max-width: 1440px) {
    .ConteudoBlogTitulo{
        font-size: 48px;  
        width: 100%;
    }
    .ConteudoBlogTexto{
        font-size: 16px;  
    }
}

/* XLARGE */
@media screen and (min-width: 1920px) {
    .ConteudoBlogTitulo{
        font-size: 80px;  
    }
    .ConteudoBlogTexto{
        font-size: 32px;  
    }
}
