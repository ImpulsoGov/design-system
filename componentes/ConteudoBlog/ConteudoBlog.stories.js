import React from 'react'
import { ConteudoBlog } from './ConteudoBlog'


export default {
  title: "Componentes/ConteudoBlog",
  component: ConteudoBlog,
  argTypes: { 
    titulo: {
      name: "titulo",
      description: "Título da página *string*"
     },
     texto: {
      name: "texto",
      description: "Parágrafo de texto *string*"
     },
     capa: {
      name: "capa",
      description: "Link da imagem de capa *URL*",
     },
     autor: {
      name: "autor",
      description: "Array de objetos contendo informações sobre o autor do texto *array* \n\n **avatar:** URL da imagem do autor da postagem *URL*, \n\n **nome:** nome do autor da postagem *string*, \n\n **data:** data da postagem do conteúdo *string*."
     },    
  },
};
const Template = (args) => <ConteudoBlog {...args}/>
export const Default = Template.bind({});
Default.args={
  titulo : "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
  texto : "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut volutpat velit nec erat aliquam fringilla. Nam vestibulum eu neque non convallis. Pellentesque elementum dignissim ante.</p><img src=\"https://media.graphassets.com/W10ybMgASFuK4fdT84GM\" alt=\"banner-20.png\" title=\"banner-20.png\" width=\"1400\" height=\"817\" /><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut volutpat velit nec erat aliquam fringilla. Nam vestibulum eu neque non convallis. Pellentesque elementum dignissim ante.</p><iframe\n          width=\"560\"\n          height=\"315\"\n          frameBorder=\"0\"\n          allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\"\n          allowFullScreen\n          title=\"https://www.youtube.com/embed/vGN5pBDaAaQ\"\n          src=\"https://www.youtube.com/embed/vGN5pBDaAaQ\"\n        ></iframe><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut volutpat velit nec erat aliquam fringilla. Nam vestibulum eu neque non convallis. Pellentesque elementum dignissim ante.</p><p></p><iframe\n          width=\"600\"\n          height=\"800\"\n          frameBorder=\"0\"\n          allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\"\n          allowFullScreen\n          title=\"https://datastudio.google.com/embed/reporting/12fb288f-4955-4930-b091-63da3f846c51/page/p_wbmkd7yvuc\"\n          src=\"https://datastudio.google.com/embed/reporting/12fb288f-4955-4930-b091-63da3f846c51/page/p_wbmkd7yvuc\"\n        ></iframe><p></p>",
  capa : "https://raw.githubusercontent.com/ImpulsoGov/design-system/6f4e11d8ebf849a76c2c3d641105449aba699a00/componentes/estatico/imagem-fundo.png",
  autor : {avatar: "https://media.graphassets.com/53A911TkTVeKdKeLllnS", nome: "IMPULSO GOV", data:"  ·  Maio 2022"},
}
