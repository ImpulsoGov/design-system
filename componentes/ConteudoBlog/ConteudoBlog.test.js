import { render, screen } from "@testing-library/react";
import { ConteudoBlog } from "./ConteudoBlog";

const COMPONENT = "ConteudoBlog";
const scenarios = [
  {
    titulo: "Lorem ipsum",
    texto: "Lorem ipsum dolor sit amet.",
    capa: "https://media.graphassets.com/53A911TkTVeKdKeLllnS",
    autor: {
      avatar: "https://media.graphassets.com/53A911TkTVeKdKeLllnS",
      nome: "Carmen",
      data: "Abril 2024",
    },
  }
];

describe(`Componente: ${COMPONENT}`, () => {
  it("deve renderizar corretamente", () => {
    render(<ConteudoBlog {...scenarios[0]} />);

    const element = screen.getByTestId(COMPONENT);

    expect(element).toMatchSnapshot();
  });
});
