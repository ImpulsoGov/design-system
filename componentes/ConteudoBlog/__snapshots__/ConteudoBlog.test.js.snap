// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente: ConteudoBlog deve renderizar corretamente 1`] = `
<div
  class="ConteudoBlogContainer"
  data-testid="ConteudoBlog"
>
  <div>
    <div
      class="ConteudoBlogTitulo"
    >
      Lorem ipsum
    </div>
    <img
      class="ConteudoBlogCapa"
      src="https://media.graphassets.com/53A911TkTVeKdKeLllnS"
    />
  </div>
  <div
    class="ConteudoBlogAutor"
  >
    <img
      alt="https://media.graphassets.com/53A911TkTVeKdKeLllnS"
      class="ConteudoBlogAutorAvatar"
      src="https://media.graphassets.com/53A911TkTVeKdKeLllnS"
    />
    <div>
      Carmen
       
      <span
        class="ConteudoBlogData"
      >
          Â·  
        Janeiro
         
        2024
      </span>
    </div>
  </div>
  <div
    class="ConteudoBlogTexto"
  >
    Lorem ipsum dolor sit amet.
  </div>
</div>
`;
